test/parallel/test-http-mutable-headers.js

```diff
@@ -79,8 +79,7 @@ const s = http.createServer(common.mustCall((req, res) => {
         'set-cookie': cookies,
         'x-test-array-header': arrayValues
       });
-      // eslint-disable-next-line no-restricted-properties
-      assert.deepEqual(headersCopy['set-cookie'], cookies);
+      assert.deepStrictEqual(headersCopy['set-cookie'], cookies);
```

Comment:
I want to suggest
```js
      const headersCopy = res.getHeaders();
      // eslint-disable-next-line no-restricted-properties
      const expected = {
        'x-test-header': 'testing',
        'x-test-header2': 'testing',
        'set-cookie': cookies,
        'x-test-array-header': arrayValues
      };
      Object.setPrototypeOf(expected, null)
      assert.deepStrictEqual(headersCopy, expected);
```

And a question, why is `headersCopy.__proto__ === undifined`?
Also `headersCopy.hasOwnPropert === undifined` so the API is wrong it says the return value is an `Object`
https://nodejs.org/api/http.html#http_response_getheaders